File: Java/Tests/jdi-uitest-webtests/src/test/java/com/epam/jdi/uitests/testing/unittests/tests/complex/SelectorTests.java
package com.epam.jdi.uitests.testing.unittests.tests.complex;

import com.epam.jdi.uitests.testing.unittests.InitTests;
import com.epam.jdi.uitests.testing.unittests.enums.Odds;
import com.epam.jdi.uitests.web.selenium.elements.complex.Selector;
import org.testng.annotations.BeforeMethod;
import org.testng.annotations.Test;

import java.io.IOException;
import java.lang.reflect.Method;
import java.util.List;

import static com.epam.jdi.uitests.testing.unittests.enums.Odds.SEVEN;
import static com.epam.jdi.uitests.testing.unittests.enums.Preconditions.METALS_AND_COLORS_PAGE;
import static com.epam.jdi.uitests.testing.unittests.pageobjects.EpamJDISite.metalsColorsPage;
import static com.epam.jdi.uitests.testing.unittests.tests.complex.CommonActionsData.*;
import static com.epam.jdi.uitests.core.preconditions.PreconditionsState.isInState;
import static com.epam.web.matcher.testng.Assert.areEquals;
import static com.epam.web.matcher.testng.Assert.listEquals;
import static java.util.Arrays.asList;

/**
 * Created by Roman_Iovlev on 9/15/2015.
 */
public class SelectorTests extends InitTests {
    private static final List<String> oddOptions = asList("1", "3", "5", "7");

    private Selector<Odds> odds() {
        return metalsColorsPage.summary.odds;
    }

    @BeforeMethod
    public void before(Method method) throws IOException {
        isInState(METALS_AND_COLORS_PAGE, method);
    }

    @Test
    public void selectStringTest() {
        odds().select("7");
        checkAction("Summary (Odd): value changed to 7");
    }

    @Test
    public void selectIndexTest() {
        odds().select(4);
        checkAction("Summary (Odd): value changed to 7");
    }

    @Test
    public void selectEnumTest() {
        odds().select(SEVEN);
        checkAction("Summary (Odd): value changed to 7");
    }

    @Test
    public void getOptionsTest() {
        listEquals(odds().getOptions(), oddOptions);
    }

    @Test
    public void getNamesTest() {
        listEquals(odds().getNames(), oddOptions);
    }

    @Test
    public void getValuesTest() {
        listEquals(odds().getValues(), oddOptions);
    }

    @Test
    public void getOptionsAsTextTest() {
        areEquals(odds().getOptionsAsText(), "1, 3, 5, 7");
    }

    @Test
    public void setValueTest() {
        odds().setValue("7");
        checkAction("Summary (Odd): value changed to 7");
    }

    @Test
    public void getNameTest() {
        areEquals(odds().getName(), "Odds");
    }

    // Fails
    @Test
    public void getSelectedTest() {
        checkActionThrowError(() -> odds().getSelected(), noElementsMessage); // isDisplayed not defined
    }

    @Test
    public void getSelectedIndexTest() {
        checkActionThrowError(() -> odds().getSelectedIndex(), noElementsMessage); // isDisplayed not defined
    }

    @Test
    public void isSelectedTest() {
        checkActionThrowError(() -> odds().isSelected("7"), noElementsMessage); // isDisplayed not defined
    }

    @Test
    public void isSelectedEnumTest() {
        checkActionThrowError(() -> odds().isSelected(SEVEN), noElementsMessage); // isDisplayed not defined
    }

    @Test
    public void waitSelectedTest() {
        checkActionThrowError(() -> odds().waitSelected("7"), noElementsMessage); // isDisplayed not defined
    }

    @Test
    public void waitSelectedEnumTest() {
        checkActionThrowError(() -> odds().waitSelected(SEVEN), noElementsMessage); // isDisplayed not defined
    }

    @Test
    public void getValueTest() {
        checkActionThrowError(() -> odds().getValue(), noElementsMessage); // isDisplayed not defined
    }
}

File: Python/Test/jdi_uitests_webtests/test/complex/selector_tests.py
from JDI.jdi_assert.testing.assertion import Assert
from Test.jdi_uitests_webtests.main.enums.entities import Odds
from Test.jdi_uitests_webtests.main.enums.preconditions import Preconditions
from Test.jdi_uitests_webtests.main.page_objects.epam_jdi_site import EpamJDISite
from Test.jdi_uitests_webtests.main.utils.common_action_data import CommonActionsData
from Test.jdi_uitests_webtests.test.init_tests import InitTests


class SelectorTests(InitTests):
    odd_options = ["1", "3", "5", "7"]

    radio_buttons = EpamJDISite.metals_colors_page.summary.odds_selector

    def setUp(self):
        super(SelectorTests, self).setUp(self.id().split(".")[-1])
        Preconditions.METALS_AND_COLORS_PAGE.is_in_state()

    def test_select_string(self):
        self.radio_buttons.select("7")
        CommonActionsData.check_action("Summary (Odd): value changed to 7")

    def test_select_index(self):
        self.radio_buttons.select(4)
        CommonActionsData.check_action("Summary (Odd): value changed to 7")

    def test_select_enum(self):
        self.radio_buttons.select(Odds.SEVEN)
        CommonActionsData.check_action("Summary (Odd): value changed to 7")

    def test_get_options(self):
        Assert.assert_equal(self.radio_buttons.get_options(), self.odd_options)

    def test_get_names(self):
        Assert.assert_equal(self.radio_buttons.get_names(), self.odd_options)

    def test_get_values(self):
        Assert.assert_equal(self.radio_buttons.get_values(), self.odd_options)

    def test_get_options_as_text(self):
        Assert.assert_equal(self.radio_buttons.get_options_as_text(), ", ".join(self.odd_options))

    def test_set_value_text(self):
        self.radio_buttons.set_value("7")
        CommonActionsData.check_action("Summary (Odd): value changed to 7")

    def test_get_name(self):
        Assert.assert_equal(self.radio_buttons.get_name(),"odds_selector")

    def test_get_selected(self):
        Assert.assert_equal(self.radio_buttons.get_selected(), "1")

    def test_get_selected_index(self):
        Assert.assert_equal(self.radio_buttons.get_selected_index(), 0)

    def test_is_selected(self):
        Assert.assert_equal(self.radio_buttons.is_selected("7"), False)
        Assert.assert_equal(self.radio_buttons.is_selected("1"), True)

    def test_is_selected_enum(self):
        Assert.assert_equal(self.radio_buttons.is_selected(Odds.SEVEN), False)
        Assert.assert_equal(self.radio_buttons.is_selected(Odds.ONE), True)

    def test_get_value(self):
        Assert.assert_equal(self.radio_buttons.get_value(), "1")